Code/Engine/Core/System/Implementation/android/InputDevice_android.inl

```diff
@@ -0,0 +1,193 @@
+  {
+    // Update pointer positions
+    const ezUInt64 uiPointerCount = AMotionEvent_getPointerCount(pEvent);
+    for (ezUInt32 uiPointerIndex = 0; uiPointerIndex < uiPointerCount; uiPointerIndex++)
+    {
+      const float fPixelX = AMotionEvent_getX(pEvent, uiPointerIndex);
+      const float fPixelY = AMotionEvent_getY(pEvent, uiPointerIndex);
+      const ezInt32 id = AMotionEvent_getPointerId(pEvent, uiPointerIndex);
+      if (id < 10)
+      {
+        m_InputSlotValues[ezInputManager::GetInputSlotTouchPointPositionX(id)] = static_cast<float>(fPixelX / static_cast<float>(screens[0].m_iResolutionX));
+        m_InputSlotValues[ezInputManager::GetInputSlotTouchPointPositionY(id)] = static_cast<float>(fPixelY / static_cast<float>(screens[0].m_iResolutionY));
+        DEBUG_LOG("Finger MOVE: {} = {} x {}", id, m_InputSlotValues[ezInputManager::GetInputSlotTouchPointPositionX(id)], m_InputSlotValues[ezInputManager::GetInputSlotTouchPointPositionY(id)]);
+      }
+    }
+
+    // Update pointer state
+    const ezUInt32 uiActionEvent = uiAction & AMOTION_EVENT_ACTION_MASK;
+    const ezUInt32 uiActionPointerIndex = (uiAction & AMOTION_EVENT_ACTION_POINTER_INDEX_MASK) >> AMOTION_EVENT_ACTION_POINTER_INDEX_SHIFT;
+
+    const ezInt32 id = AMotionEvent_getPointerId(pEvent, uiActionPointerIndex);
+    // We only support up to 10 touch points at the same time.
+    if (id >= 10)
+      return 0;
+
+    {
+      // Not sure if the action finger is always present in the upper loop of uiPointerCount, so we update it here for good measure.
+      const float fPixelX = AMotionEvent_getX(pEvent, uiActionPointerIndex);
+      const float fPixelY = AMotionEvent_getY(pEvent, uiActionPointerIndex);
+      m_InputSlotValues[ezInputManager::GetInputSlotTouchPointPositionX(id)] = static_cast<float>(fPixelX / static_cast<float>(screens[0].m_iResolutionX));
+      m_InputSlotValues[ezInputManager::GetInputSlotTouchPointPositionY(id)] = static_cast<float>(fPixelY / static_cast<float>(screens[0].m_iResolutionY));
+      DEBUG_LOG("Finger MOVE: {} = {} x {}", id, m_InputSlotValues[ezInputManager::GetInputSlotTouchPointPositionX(id)], m_InputSlotValues[ezInputManager::GetInputSlotTouchPointPositionY(id)]);
+    }
+
+    switch (uiActionEvent)
+    {
+      case AMOTION_EVENT_ACTION_DOWN:
+      case AMOTION_EVENT_ACTION_POINTER_DOWN:
+        m_InputSlotValues[ezInputManager::GetInputSlotTouchPoint(id)] = 1;
+        DEBUG_LOG("Finger DOWN: {}", id);
+        return 1;
+      case AMOTION_EVENT_ACTION_MOVE:
+        // Finger moved (we always update that at the top).
+        return 1;
+      case AMOTION_EVENT_ACTION_UP:
+      case AMOTION_EVENT_ACTION_POINTER_UP:
+      case AMOTION_EVENT_ACTION_CANCEL:
+      case AMOTION_EVENT_ACTION_OUTSIDE:
+        m_InputSlotValues[ezInputManager::GetInputSlotTouchPoint(id)] = 0;
+        DEBUG_LOG("Finger UP: {}", id);
+        return 1;
+      case AMOTION_EVENT_ACTION_SCROLL:
```

Comment:
We have a mouse wheel event, that could be used for the scrolling.

Reply:
Well, I implemented it now, but it looks like it's actually only used for an actual wheel on a physical mouse, not finger scrolling. But that's half a mouse already implemented right?🙂